### PR TITLE
#68-feat: searchbar

### PR DESCRIPTION
**Pull request by Almafe2510**

### CODE REVIEW

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  },
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
+  display: 'flex',
+  alignItems: 'center',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:30]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  alignItems: "center",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+    marginLeft: theme.spacing(1),
+    width: 'auto',
+  },
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:28]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  pointerEvents: "none",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import InputBase from '@mui/material/InputBase';
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:6]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
import InputBase from "@mui/material/InputBase";
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+const Search = styled('div')(({ theme }) => ({
+  position: 'relative',
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  '&:hover': {
+    backgroundColor: alpha(theme.palette.common.white, 0.25),
+  },
+  marginLeft: 0,
+  width: '100%',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:17]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  width: "100%",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 5]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import AppBar from '@mui/material/AppBar';
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:3]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
import AppBar from "@mui/material/AppBar";
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 6]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:2]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
import { styled, alpha } from "@mui/material/styles";
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+    color: 'inherit',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:35]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
    color: "inherit",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:1]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
import React, { useState } from "react";
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:31]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  justifyContent: "center",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+    color: 'inherit',
+    '& .MuiInputBase-input': {
+      padding: theme.spacing(1, 1, 1, 0),
+      // vertical padding + font size from searchIcon
+      paddingLeft: `calc(1em + ${theme.spacing(4)})`,
+      transition: theme.transitions.create('width'),
+      width: '100%',
+      [theme.breakpoints.up('sm')]: {
+        width: '12ch',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:43]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
        width: "12ch",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 11]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import InputBase from '@mui/material/InputBase';
+import SearchIcon from '@mui/icons-material/Search';
+
+const Search = styled('div')(({ theme }) => ({
+  position: 'relative',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:10]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  position: "relative",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+    width: 'auto',
+  },
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: '100%',
+  position: 'absolute',
+  pointerEvents: 'none',
+  display: 'flex',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:29]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  display: "flex",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  '&:hover': {
+    backgroundColor: alpha(theme.palette.common.white, 0.25),
+  },
+  marginLeft: 0,
+  width: '100%',
+  [theme.breakpoints.up('sm')]: {
+    marginLeft: theme.spacing(1),
+    width: 'auto',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:20]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
    width: "auto",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    '& .MuiInputBase-input': {
+      padding: theme.spacing(1, 1, 1, 0),
+      // vertical padding + font size from searchIcon
+      paddingLeft: `calc(1em + ${theme.spacing(4)})`,
+      transition: theme.transitions.create('width'),
+      width: '100%',
+      [theme.breakpoints.up('sm')]: {
+        width: '12ch',
+        '&:focus': {
+          width: '20ch',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:45]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
          width: "20ch",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 15]
```diff
@@ -0,0 +1,96 @@
...
+  justifyContent: 'center',
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+    color: 'inherit',
+    '& .MuiInputBase-input': {
+      padding: theme.spacing(1, 1, 1, 0),
+      // vertical padding + font size from searchIcon
+      paddingLeft: `calc(1em + ${theme.spacing(4)})`,
+      transition: theme.transitions.create('width'),
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:40]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
      transition: theme.transitions.create("width"),
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 16]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import InputBase from '@mui/material/InputBase';
+import SearchIcon from '@mui/icons-material/Search';
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:7]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
import SearchIcon from "@mui/icons-material/Search";
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 17]
```diff
@@ -0,0 +1,96 @@
...
+  },
+  marginLeft: 0,
+  width: '100%',
+  [theme.breakpoints.up('sm')]: {
+    marginLeft: theme.spacing(1),
+    width: 'auto',
+  },
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:24]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
const SearchIconWrapper = styled("div")(({ theme }) => ({
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+  padding: theme.spacing(0, 2),
+  height: '100%',
+  position: 'absolute',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:27]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  position: "absolute",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+  pointerEvents: 'none',
+  display: 'flex',
+  alignItems: 'center',
+  justifyContent: 'center',
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+    color: 'inherit',
+    '& .MuiInputBase-input': {
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:36]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
    "& .MuiInputBase-input": {
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+  width: '100%',
+  [theme.breakpoints.up('sm')]: {
+    marginLeft: theme.spacing(1),
+    width: 'auto',
+  },
+}));
+
+const SearchIconWrapper = styled('div')(({ theme }) => ({
+  padding: theme.spacing(0, 2),
+  height: '100%',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:26]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  height: "100%",
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import InputBase from '@mui/material/InputBase';
+import SearchIcon from '@mui/icons-material/Search';
+
+const Search = styled('div')(({ theme }) => ({
+  position: 'relative',
+  borderRadius: theme.shape.borderRadius,
+  backgroundColor: alpha(theme.palette.common.white, 0.15),
+  '&:hover': {
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:13]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  "&:hover": {
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 22]
```diff
@@ -0,0 +1,96 @@
+import React, { useState } from 'react';
+import { styled, alpha } from '@mui/material/styles';
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import InputBase from '@mui/material/InputBase';
+import SearchIcon from '@mui/icons-material/Search';
+
+const Search = styled('div')(({ theme }) => ({
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
const Search = styled("div")(({ theme }) => ({
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+    color: 'inherit',
+    '& .MuiInputBase-input': {
+      padding: theme.spacing(1, 1, 1, 0),
+      // vertical padding + font size from searchIcon
+      paddingLeft: `calc(1em + ${theme.spacing(4)})`,
+      transition: theme.transitions.create('width'),
+      width: '100%',
+      [theme.breakpoints.up('sm')]: {
+        width: '12ch',
+        '&:focus': {
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:44]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
        "&:focus": {
```

File: frontend/src/components/molecules/SearchBar/SearchBar.tsx [hunk 24]
```diff
@@ -0,0 +1,94 @@
...
+}));
+
+const StyledInputBase = styled(InputBase)(({ theme }) => ({
+    color: "inherit",
+    "& .MuiInputBase-input": {
+      padding: theme.spacing(1, 1, 1, 0),
+      // vertical padding + font size from searchIcon
+      paddingLeft: "calc(1em + ${theme.spacing(4)})",
+      transition: theme.transitions.create("width"),
+      width: '100%',
```
Comment (JSnow11) [inline frontend/src/components/molecules/SearchBar/SearchBar.tsx:39]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/ISPP-Eventia/eventus/pullRequest?prid=9096486)
```suggestion
  width: "100%",
```

**Review by laur2000 — APPROVED** [review]

**Pull request closed without merging**